documentation/src/main/asciidoc/user_guide/integrations.adoc

```diff
@@ -11,6 +11,214 @@ Also check the link:$$https://github.com/infinispan/infinispan-spark/tree/master
+----
+
+==== Loading an existing Lucene Index
+
+The _org.infinispan.lucene.cachestore.LuceneCacheLoader_ is an Infinispan CacheLoader able to have Infinispan directly load data from an existing Lucene index into the grid. Currently this supports reading only.
+
+[options="header"]
+|===============
+|Property|Description|Default
+| _location_ |The path where the indexes are stored. Subdirectories (of first level only) should contain the indexes to be loaded, each directory matching the index name attribute of the InfinispanDirectory constructor.|none (mandatory)
+| _autoChunkSize_ |A threshold in bytes: if any segment is larger than this, it will be transparently chunked in smaller cache entries up to this size.|32MB
+
+|===============
+
+It's worth noting that the IO operations are delegated to Lucene's standard _org.apache.lucene.store.FSDirectory_ , which will select an optimal approach for the running platform.
+
+Implementing write-through should not be hard: you're welcome to try implementing it.
+
+
+==== Architectural limitations
+This Directory implementation makes it possible to have almost real-time reads across multiple nodes. A fundamental limitation of the Lucene design is that only a single IndexWriter is allowed to make changes on the index: a pessimistic lock is acquired by the writer; this is generally ok as a single IndexWriter _instance_ is very fast and accepts update requests from multiple threads. When sharing the Directory across Infinispan nodes the IndexWriter limitation is not lifted: since you can have only one instance, that reflects in your application as having to apply all changes on the same node. There are several strategies to write from multiple nodes on the same index:
+
+.Index write strategies
+* One node writes, the other delegate to it sending messages
+* Each node writes on turns
+* You application makes sure it will only ever apply index writes on one node
+
+The _Infinispan Lucene Directory_ protects its content by implementing a distributed locking strategy, though this is designed as a last line of defense and is not to be considered an efficient mechanism to coordinate multiple writes: if you don't apply one of the above suggestions and get high write contention from multiple nodes you will likely get timeout exception.
+
+==== Suggestions for optimal performance
+
+===== JGroups and networking stack
+JGroups manages all network IO and as such it is a critical component to tune for your specific environment. Make sure to read the link:$$http://jgroups.org/manual-3.x/html/index.html$$[JGroups reference documentation] , and play with the performance tests included in JGroups to make sure your network stack is setup appropriately. Don't forget to check also operating system level parameters, for example buffer sizes dedicated for networking. JGroups will log warning when it detects something wrong, but there is much more you can look into.
```

Comment:
Yes, but soon can be on `4.0.0.Alpha1` - https://github.com/infinispan/infinispan/pull/4529. I would like to make sure @danberindei and You are in sync with this...